src/compression/snappy.rs

```diff
@@ -14,7 +14,7 @@ pub fn compress(src: &[u8]) -> Result<Vec<u8>> {
                  buf.truncate(len);
                  buf
              })
-        .map_err(|err| Error::InvalidSnappy(err))
+        .map_err(|err| ErrorKind::InvalidSnappy(err).into())
```

Comment:
unfortunately, `bail!` always `return Err(_)` instead of `Error`

```rust
#[macro_export]
macro_rules! bail {
    ($e:expr) => {
        return Err($e.into());
    };
    ($fmt:expr, $($arg:tt)+) => {
        return Err(format!($fmt, $($arg)+).into());
    };
}
```